### DISCUSSION TITLE
Using variables in a query

### DISCUSSION THREAD

**Discussion by melohagan** [sidorares/node-mysql2#1636] [Q&A] [answered]
Hello!

I am wondering if it's possible to use variables in the mysql2 **query** method?

I can successfully run this in MySQL Workbench:
```sql
set @custNum = 103;
SELECT * FROM customers WHERE customerNumber = @custNum;
``` 

but when run via `client.query(...)` I'm receiving the following error:
```
code: 'ER_PARSE_ERROR',
errno: 1064,
sql: 'set @custNum = 103; SELECT * FROM customers WHERE customerNumber = @custNum;',
sqlState: '42000',
sqlMessage: "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'SELECT * FROM customers WHERE customerNumber = @custNum' at line 1"
```

Related discussion: https://github.com/Budibase/budibase/discussions/7775

**Comment by sidorares** [comment, marked as answer]
do you have `multipleStatements` connection flag enabled? See https://github.com/mysqljs/mysql#multiple-statement-queries
At a protocol level there is nothing special in variables, query with variable in it is sent using the same COM_QUERY command, its just part of a query text

**Reply by melohagan** [reply]
Thanks for the fast reply! Enabling that flag worked. I'll have to confer with the rest of the team to see if this will be an issue:
 > it allows for SQL injection attacks if values are not properly escaped

**Comment by sidorares** [comment]
Some unit tests are using variables in the query, example: https://github.com/sidorares/node-mysql2/blob/d065991c88c98aa028420147f7b005ebd76f5449/test/integration/connection/test-multiple-results.js#L77
